src/components/PopupWithForm.js

```diff
@@ -20,14 +30,29 @@ export default class PopupWithForm extends Popup {
     return fromValues;
   }
 
-  setEventListners() {
-    super.setEventListners();
-    this._popupForm.addEventListener("submit", (e) => {
-      this._handleFormSubmit(this._getinputvalues());
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit",  (e) => {
+      e.preventDefault();
+      this.setLoading(true);
+      this._submitButton.disabled = true;
+      const formData = this._getinputvalues();
+      //setTimeout(() => {
+       this._handleFormSubmit(formData);
+      //  this._cardFormSubmit(formData);
+      //  this._submitButton.textContent = this._defaultButtonText;
+      //  this._submitButton.disabled = false;
+      //} 3000);
       this._popupForm.reset();
     });
   }
+
+  setSubmitAction(handleSubmit, cardSubmit) {
+    this._handleFormSubmit = handleSubmit;
+    this._cardFormSubmit= cardSubmit;
```

Comment:
Please, delete `cardSubmit` and `this._cardFormSubmit= cardSubmit;` because you don't use it, and you can't use it in `PopupWithForm`